pkg/lib/operatorstatus/builder.go

```diff
@@ -0,0 +1,152 @@
+package operatorstatus
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/clock"
+)
+
+// NewBuilder returns a builder for ClusterOperatorStatus.
+func NewBuilder(clock clock.Clock) *Builder {
+	return &Builder{
+		clock: clock,
+	}
+}
+
+// Builder helps build ClusterOperatorStatus with appropriate
+// ClusterOperatorStatusCondition and OperandVersion.
+type Builder struct {
+	clock  clock.Clock
```

Comment:
Nice! This seems cleaner than the `timeNow(...)`. Maybe we should do this for our `Operator`s as well.